### PR TITLE
Override native default label if config server default label is not null

### LINKED ISSUE #1184: Property spring.cloud.config.server.native.default-label is being ignored
The value of `spring.cloud.config.server.native.default-label`, which should default to `master` is always overwritten by the value of `spring.cloud.config.server.default-label`.

If `server.default-label` is non-null, this behaviour is correct and consistent with SCM-based repositories, but in case `server.default-label` is null, the native repository's setting should not be overwritten.

The Git and SVN repository factories check `server.default-label` and only set the default label on the created repository if it is non-null.

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Fixes #1184

### CODE REVIEW

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		NativeEnvironmentRepositoryFactory factory = new NativeEnvironmentRepositoryFactory(new StandardEnvironment(), props);
+		NativeEnvironmentProperties environmentProperties = new NativeEnvironmentProperties();
+		NativeEnvironmentRepository repo = factory.build(environmentProperties);
+		assertEquals("mylabel", repo.getDefaultLabel());
+
+		factory = new NativeEnvironmentRepositoryFactory(new StandardEnvironment(), props);
+		environmentProperties = new NativeEnvironmentProperties();
+		environmentProperties.setDefaultLabel("mynewlabel");
+		repo = factory.build(environmentProperties);
+		assertEquals("mylabel", repo.getDefaultLabel());
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: Shouldn't this be "mynewlabel"?

Comment (spencergibb) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: I guess I'm confused as server default label has precedence over the repo specific one.

Comment (ryanjbaxter) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: I thought so too ;)  But the way SVN and GIT repos work today is the config server default label property overrides the environment repo default label property as long as its not `null`.

https://github.com/spring-cloud/spring-cloud-config/blob/2.0.x/spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/SvnEnvironmentRepositoryFactory.java

Comment (spencergibb) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: :-) but why?

Comment (ryanjbaxter) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: its a good question, i dont see where we have documented the correct behavior, my worry about changing it now would be breaking people relying on it.

Comment (spencergibb) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: oh well, best make it consistent.

Comment (sunruh) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: > :-) but why?

@spencergibb They (SVN & Git) only let the config server default override the repositories default if it's non-null as SVN defaults to `trunk` and Git to `master`. So when someone is not using labels at all (and not setting the config server default), they behave correctly and use their implementation-specific default or the default set in their specific configuration. That way one can set the default globally (taking precedence) or on a per-repository basis.

Comment (spencergibb) [reply spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepositoryFactoryTest.java]: I'm not questioning consistency. I'm questioning why the global takes precedence over the specific. Seems backward to me, but that's for another issue.

**Pull request merged**